src/core/packages/saved-objects/api-server-internal/src/lib/apis/find.ts

```diff
@@ -170,10 +170,15 @@ export const performFind = async <T = unknown, A = unknown>(
       // If the user is unauthorized to find *anything* they requested, return an empty response
       return SavedObjectsUtils.createEmptyFindResponse<T, A>(options);
     }
-    if (authorizationResult?.status === 'partially_authorized') {
+    if (
+      authorizationResult?.status === 'fully_authorized' ||
+      authorizationResult?.status === 'partially_authorized'
+    ) {
       typeToNamespacesMap = new Map<string, string[]>();
       for (const [objType, entry] of authorizationResult.typeMap) {
         if (!entry.find) continue;
+        // Discard the types that the SO repository doesn't know about (typically hidden objects).
+        if (!allowedTypes.includes(objType)) continue;
```

Comment:
Mixing the user's authorization and the client's scope here (which can be odd). 

However, `typeToNamespacesMap` doesn't really imply that it's bound to only one.